docs/source/tutorials/custom_environments.rst

```diff
@@ -0,0 +1,293 @@
+            # also reward on mission end
+            handlers.RewardForMissionEnd(50)
+        ]
+
+Construct a Quit Handler
+====================================
+We want the episode to terminate when the agent obtains a gold block.
+
+.. code-block:: python 
+
+    def create_agent_handlers(self) -> List[Handler]:
+        return [
+            # make the agent quit when it gets a gold block in its inventory
+            handlers.AgentQuitFromPossessingItem([
+                dict(type="gold_block", amount=1)
+            ])
+        ]
+
+Allow the Agent to Place Water
+====================================
+We want the agent to be able to place the water bucket, but :code:`SimpleEmbodimentEnvSpec`
+does not provide this ability by default. Note that we call :code:`super().create_actionables()`
+so that we keep the actions which :code:`SimpleEmbodimentEnvSpec` does provide by default (like movement, jumping)
+
+
+.. code-block:: python
+
+    def create_actionables(self) -> List[Handler]:
+        return super().create_actionables() + [
+            # allow agent to place water
+            handlers.KeybasedCommandAction("use"),
+            # also allow it to equip the pickaxe
+            handlers.EquipAction(["diamond_pickaxe"])
+        ]
+
+Give Extra Observations
+====================================
+In addition to the POV image data the agent receives as an observation, lets provide
+it with compass and lifestats data. We override :code:`create_observables` just like the previous step.
+
+.. code-block:: python
+
+    def create_observables(self) -> List[Handler]:
+        return super().create_observables() + [
+            # current location and lifestats are returned as additional
+            # observations
+            handlers.ObservationFromCurrentLocation(),
+            handlers.ObservationFromLifeStats()
+        ]
+
+Set the Time 
+======================
+Lets set the time to morning.
+
+.. code-block:: python
+
+    def create_server_initial_conditions(self) -> List[Handler]:
+        return [
+            # Sets time to morning and stops passing of time
+            handlers.TimeInitialCondition(False, 23000)
+        ]
+
+Other Functions to Implement
+====================================
+
+:code:`SimpleEmbodimentEnvSpec` requires that we implement these methods.
+
+.. code-block:: python
+
+    # see API reference for use cases of these first two functions
+
+    def create_server_quit_producers(self):
+        return []
+    
+    def create_server_decorators(self) -> List[Handler]:
+        return []
+
+    # the episode can terminate when this is True
+    def determine_success_from_rewards(self, rewards: list) -> bool:
+        return sum(rewards) >= self.reward_threshold
+
+    def is_from_folder(self, folder: str) -> bool:
+        return folder == 'mlgwb'
+
+    def get_docstring(self):
+        return MLGWB_DOC
+
+**Congrats!** You just made your first MineRL environment. Checkout the herobraine API reference 
```

Comment:
I mention a link to that at the top of the tutorial: https://github.com/trigaten/MLGPK_gym
Ill add another link there. Would you prefer I add the file(s) to MineRL rather than my Github?

Reply:
Hmm I would prefer having the example code in the MineRL repo (easier to modify by others etc), so I would like to see the complete python code of this example somewhere in the MineRL repo (say, a new directory in the root named `examples` and this example file in there). In addition to pointing to this file you can also point to your repo (e.g. if it contains ipynb versions or more details) :)